bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/CallOptionsFactory.java

```diff
@@ -48,17 +63,29 @@
    * <p>For internal use only - public for technical reasons.
    */
   @InternalApi("For internal usage only")
-  public static class Default implements CallOptionsFactory {
+  class Default implements CallOptionsFactory {
```

Comment:
Not sure why this was a class before. Maybe we don't need to have separate classes for default and configured behaviors? CallOptionFactory could return default values if CallOptionsConfig is not provided?

Reply:
I see both sides of this; if the main implementation is simple, we may not need a default one. In this case, I can somewhat see why a default impl was used, since it didn't use the configurations.

Having said that - I refactored this whole thing quite a bit, as we wanted the CallOptionsFactory to handle more of the RPC and Deadline logic that was originally in AbstractRetryOperations. So I now do definitely have a separate default and main implementation.

(And they are now in separate files; this file was already too big as it was :) )

Reply:
I agree with Mattie, the Defaults arent ever actually used in production code. So I think we should remove them. However, I think that will create a lot more noise in PR and I dont want to block this feature. So I think we should just ship it as is

Reply:
FYI, we do use this in BigtableTableAdminGrpcClient. There are no CallOptionsConfig defined for that class.
While it wouldn't be hard to inject it, I don't know what the implications would be to have it.
(If we trust our existing test suites to handle this change, then it would be quick to do this)